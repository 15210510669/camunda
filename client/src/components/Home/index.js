/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
