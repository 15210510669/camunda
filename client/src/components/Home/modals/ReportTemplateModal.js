/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import {t} from 'translation';

import TemplateModal from './TemplateModal';

import heatmapImg from './images/heatmap.jpg';
import durationImg from './images/duration.svg';
import tableImg from './images/table.svg';
import chartImg from './images/chart.svg';

import './ReportTemplateModal.scss';

export default function ReportTemplateModal({onClose}) {
  const templates = [
    {name: 'blank'},
    {
      name: 'heatmap',
      img: heatmapImg,
      config: {
        view: {entity: 'flowNode', properties: ['frequency']},
        groupBy: {type: 'flowNodes', value: null},
        visualization: 'heat',
        configuration: {
          xLabel: t('report.groupBy.flowNodes'),
          yLabel: t('report.view.fn') + ' ' + t('report.view.count'),
        },
      },
    },
    {
      name: 'number',
      img: durationImg,
      config: {
        view: {entity: 'processInstance', properties: ['duration']},
        groupBy: {type: 'none', value: null},
        visualization: 'number',
        configuration: {
          yLabel: t('report.view.pi') + ' ' + t('report.view.duration'),
        },
      },
    },
    {
      name: 'table',
      img: tableImg,
      config: {
        view: {entity: 'userTask', properties: ['frequency']},
        groupBy: {type: 'userTasks', value: null},
        visualization: 'table',
        configuration: {
          xLabel: t('report.groupBy.userTasks'),
          yLabel: t('report.view.userTask') + ' ' + t('report.view.count'),
        },
      },
    },
    {
      name: 'chart',
      img: chartImg,
      config: {
        view: {entity: 'processInstance', properties: ['frequency']},
        groupBy: {type: 'startDate', value: {unit: 'automatic'}},
        visualization: 'bar',
        configuration: {
          xLabel: t('report.groupBy.startDate'),
          yLabel: t('report.view.pi') + ' ' + t('report.view.count'),
        },
      },
    },
  ];

  return (
    <TemplateModal
      className="ReportTemplateModal"
      onClose={onClose}
      templates={templates}
      entity="report"
      templateToState={({name, template, definitions, xml}) => ({
        name,
        data: {
          ...(template || {}),
          configuration: {...(template?.configuration || {}), xml},
          definitions,
        },
      })}
    />
  );
}
