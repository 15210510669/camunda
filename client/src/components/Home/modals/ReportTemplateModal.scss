.ReportTemplateModal {
  &.TemplateModal.noProcess {
    .DefinitionSelection,
    .diagramArea {
      visibility: visible;
    }
  }
}
