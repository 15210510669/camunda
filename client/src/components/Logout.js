/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    mightFail(
      get('api/authentication/logout'),
      () => {
        addNotification({text: t('navigation.logoutSuccess')});
        history.replace('/');
      },
      () => {
        addNotification({text: t('navigation.logoutFailed'), type: 'error'});
        history.replace('/');
      }
    );
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
