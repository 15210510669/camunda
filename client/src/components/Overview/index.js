export {default as Overview} from './Overview';
export {default as Alerts} from './Alerts';
