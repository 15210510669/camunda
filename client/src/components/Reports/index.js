export {default as Reports} from './Reports';
export {default as Report} from './Report';
export {Filter} from './filter'
