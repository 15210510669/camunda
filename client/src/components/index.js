export {Footer} from './Footer';
export {Header, LogoutButton} from './Header';
export {Home} from './Home';
export {Login} from './Login';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboard';
export {Dashboards} from './Dashboards';
export {Report} from './Report';
export {Reports} from './Reports';
