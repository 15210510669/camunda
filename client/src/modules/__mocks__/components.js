export {default as Button} from '../components/__mocks__/Button';
export {default as Input} from '../components/__mocks__/Input';
export {default as Modal} from '../components/__mocks__/Modal';
export {default as Popover} from '../components/__mocks__/Popover';
export {default as ControlGroup} from '../components/__mocks__/ControlGroup';
export {default as ShareEntity} from '../components/__mocks__/ShareEntity';
export {default as DashboardView} from '../components/__mocks__/DashboardView';
export {default as Icon} from '../components/__mocks__/Icon';
export {default as Dropdown} from '../components/__mocks__/Dropdown';
export {default as Message} from '../components/__mocks__/Message';
export {default as ReportView} from '../components/__mocks__/ReportView';
export {default as Select} from '../components/__mocks__/Select';
export {default as BPMNDiagram} from '../components/__mocks__/BPMNDiagram';
export {default as TargetValueBadge} from '../components/__mocks__/TargetValueBadge';
export {default as Table} from '../components/__mocks__/Table';
export {default as Logo} from '../components/__mocks__/Logo';
export {default as ErrorMessage} from '../components/__mocks__/ErrorMessage';
