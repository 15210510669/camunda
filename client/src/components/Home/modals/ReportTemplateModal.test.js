/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React, {runLastEffect} from 'react';
import {shallow} from 'enzyme';

import {DefinitionSelection, Button, BPMNDiagram, LabeledInput} from 'components';
import {loadProcessDefinitionXml} from 'services';

import {ReportTemplateModal} from './ReportTemplateModal';

jest.mock('react', () => {
  const outstandingEffects = [];
  return {
    ...jest.requireActual('react'),
    useEffect: (fn) => outstandingEffects.push(fn),
    runLastEffect: () => {
      if (outstandingEffects.length) {
        outstandingEffects.pop()();
      }
    },
  };
});
jest.mock('services', () => {
  return {
    ...jest.requireActual('services'),
    loadProcessDefinitionXml: jest.fn().mockReturnValue('processXML'),
  };
});

const props = {
  onClose: jest.fn(),
  mightFail: jest.fn().mockImplementation((data, cb) => cb(data)),
};

beforeEach(() => {
  loadProcessDefinitionXml.mockClear();
});

it('should show a definition selection, and a list of templates', () => {
  const node = shallow(<ReportTemplateModal {...props} />);

  expect(node.find(DefinitionSelection)).toExist();
  expect(node.find('.templateContainer').find(Button).length).toBeGreaterThan(1);
});

it('should fetch and show the bpmn diagram when a definition is selected', () => {
  const node = shallow(<ReportTemplateModal {...props} />);

  node
    .find(DefinitionSelection)
    .simulate('change', {key: 'processDefinition', versions: ['1'], tenantIds: [null]});

  runLastEffect();

  expect(loadProcessDefinitionXml).toHaveBeenCalledWith('processDefinition', '1', null);
  expect(node.find(BPMNDiagram)).toExist();
  expect(node.find(BPMNDiagram).prop('xml')).toBe('processXML');
});

it('should include the selected parameters in the link state when creating a report', () => {
  const node = shallow(<ReportTemplateModal {...props} />);

  node.find(DefinitionSelection).simulate('change', {
    key: 'processDefinition',
    versions: ['1'],
    tenantIds: [null],
    name: 'Process Definition Name',
  });

  runLastEffect();

  node.find(LabeledInput).simulate('change', {target: {value: 'Template Report Name'}});
  node.find('.templateContainer').find(Button).at(1).simulate('click');

  expect(node.find('.Button.primary').prop('disabled')).toBe(false);
  expect(node.find('.Button.primary').prop('to')).toMatchSnapshot();
});
