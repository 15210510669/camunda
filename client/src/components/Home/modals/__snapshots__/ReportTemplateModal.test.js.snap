// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should include the selected parameters in the link state when creating a report 1`] = `
Object {
  "pathname": "report/new/edit",
  "state": Object {
    "data": Object {
      "configuration": Object {
        "xLabel": "Flow Nodes",
        "xml": "processXML",
        "yLabel": "Flow Node Count",
      },
      "groupBy": Object {
        "type": "flowNodes",
        "value": null,
      },
      "processDefinitionKey": "processDefinition",
      "processDefinitionName": "Process Definition Name",
      "processDefinitionVersions": Array [
        "1",
      ],
      "tenantIds": Array [
        null,
      ],
      "view": Object {
        "entity": "flowNode",
        "property": "frequency",
      },
      "visualization": "heat",
    },
    "name": "Template Report Name",
  },
}
`;
